homeworks/11-animations/index.html

```diff
@@ -4,6 +4,124 @@
   <link rel="stylesheet" href="index.css">
 </head>
 <body>
+  <div class="container">
+    <svg width="600" height="600" fill="none">
+      <g class="hexes">
+        <!-- Hex shape taken from https://weareoutman.github.io/rounded-polygon/ -->
+        <!-- I am aware that I really should render these dynamically instead of copy-pasting, but I wanted to keep this pure HTML + CSS -->
```

Comment:
SVGs are unique in a way as you can re-use certain parts of your SVG by using `x-link` to element id.

https://css-tricks.com/use-and-reuse-everything-in-svg-even-animations/

```html
      <defs>
        <path
          id="hex"
          stroke="none"
          fill="#606060"
          d="M 300 300 m -32 -32 m27.5 3.3867513459481a10 10 0 0 1 10 0l17.712812921102 10.226497308104a10 10 0 0 1 5 8.6602540378444l0 20.452994616207a10 10 0 0 1 -5 8.6602540378444l-17.712812921102 10.226497308104a10 10 0 0 1 -10 0l-17.712812921102 -10.226497308104a10 10 0 0 1 -5 -8.6602540378444l0 -20.452994616207a10 10 0 0 1 5 -8.6602540378444"
        />
      </defs>

      <g class="hexes">
        <!-- Hex shape taken from https://weareoutman.github.io/rounded-polygon/ -->
        <use xlink:href="#hex" id="center" class="hex" />
        <use xlink:href="#hex" id="west" class="hex" transform="translate(-68, 0)" />
        <use xlink:href="#hex" id="east" class="hex" transform="translate(68, 0)" />
        <use xlink:href="#hex" id="north-west" class="hex" transform="translate(-32, -60)" />
        <use xlink:href="#hex" id="north-east" class="hex" transform="translate(32, -60)" />
        <use xlink:href="#hex" id="south-west" class="hex" transform="translate(-32, 60)" />
        <use xlink:href="#hex" id="south-east" class="hex" transform="translate(32, 60)" />
      </g>
```

Keep in mind that defining transform property inline will override your  animation transform (which is scaling), to overcome this you can individually define `scale` as property.

```css
@keyframes hexAnimation {
  0% {
    opacity: 0;
    scale: 0.5;
  }
  8% {
    opacity: 1;
    scale: 1;
  }
  50% {
    opacity: 1;
    scale: 1;
  }
  58% {
    opacity: 0;
    scale: 0.5;
  }
}
```

https://danielcwilson.com/blog/2017/10/all-the-transform-ways/

Reply:
I tried to implement this (see pushed code), but it doesn't really do the scaling animation.

Reply:
I forgot to mention all of the CSS parts.
Change the selector from `.hexes path {` to `.hexes > * {`. The elements added to DOM using keyword `use` create a `shadow-root` element that is why the selector doesn't work.

`.hexes > * {` selector will target all descendants no matter what type.

<img width="805" alt="Screenshot 2021-07-30 at 18 51 12" src="https://user-images.githubusercontent.com/364880/127678952-07fc5ae4-30ba-4340-afab-e0346711c9b7.png">